### PR TITLE
[INTERNAL] Cleanup and logging improvements

### PR DESCRIPTION
**Pull request by RandomByte**

### CODE REVIEW

**Review by flovogt — DISMISSED** [review]
Fine for me

**Comment by RandomByte** [comment]
Normal builds now look like this:

Screenshot 2022-08-01 at 15 08 16

Screenshot 2022-08-01 at 14 28 48

**Verbose:**

Screenshot 2022-08-01 at 14 30 53

There are a lot of project build combinations possible now, and I think this might be a good way to communicate what is going on. For example, some projects might not be requested but still required to be built. Some of those might already have a build manifest and others not.

File: test/lib/build/ProjectBuilder.js [hunk 1]
```diff
@@ -293,9 +297,9 @@ test.serial("build: Multiple projects", async (t) => {
 		"project.b", "project.c"
 	], "_createRequiredBuildContexts got called with correct arguments");
 
-	t.is(requiresBuildAStub.callCount, 2, "TaskRunner#requiresBuild got called twice times for library.a");
-	t.is(requiresBuildBStub.callCount, 2, "TaskRunner#requiresBuild got called twice times for library.b");
-	t.is(requiresBuildCStub.callCount, 2, "TaskRunner#requiresBuild got called twice times for library.c");
+	t.is(requiresBuildAStub.callCount, 1, "TaskRunner#requiresBuild got called twice times for library.a");
```
Comment (flovogt) [inline test/lib/build/ProjectBuilder.js:300]: TaskRunner#requiresBuild got called once for

**Review by flovogt — CHANGES_REQUESTED** [review]
Nice improvement 🥳

File: test/lib/build/ProjectBuilder.js (hunk 1 above)
Comment (RandomByte) [reply test/lib/build/ProjectBuilder.js]: Thanks! Done 👍

**Review by flovogt — APPROVED** [review]

**Pull request merged**